q2_moshpit/eggnog/_method.py

```diff
@@ -134,6 +140,59 @@ def _annotate_seed_orthologs_runner(seed_ortholog, eggnog_db, sample_label,
     subprocess.run(cmds, check=True)
 
 
+def build_custom_diamond_db(
+        sequences: ProteinSequencesDirectoryFormat,
+        taxonomy_data: NCBITaxonomyDirFmt = None,
+        threads: int = None,
+        verbose: bool = False,
+        log: bool = False,
+        file_buffer_size: int = 67108864,
+        ignore_warnings: bool = False,
+        no_parse_seqids: bool = False
+        ) -> DiamondDatabaseDirFmt:
+    '''
+    Builds diamond database from protein reference database file in FASTA
+    format.
+    '''
+    # Process input parameters
+    kwargs = {}
+    for key, value in locals().items():
+        if key not in ["sequences", "taxonomy_data", "kwargs"]:
+            kwargs[key] = value
+
+    # Add paths to taxonomy data if provided
+    if taxonomy_data is not None:
+        kwargs["taxonmap"] = os.path.join(
+            str(taxonomy_data), "prot.accession2taxid.gz"
+            )
+        kwargs["taxonnodes"] = os.path.join(str(taxonomy_data), "nodes.dmp")
+        kwargs["taxonnames"] = os.path.join(str(taxonomy_data), "names.dmp")
+
+    # Filter out all kwargs that are falsy (except 0 and 0.0)
+    parsed_args = _process_common_input_params(
+        processing_func=_parse_build_diamond_db_params, params=kwargs
+    )
+
+    # Instantiate output object
+    diamond_db = DiamondDatabaseDirFmt()
+
+    # Define path to in/output file
+    path_in = os.path.join(str(sequences), "protein-sequences.fasta")
+    path_out = os.path.join(str(diamond_db), "ref_db.dmnd")
```

Comment:
Again, you're only using those paths below - you may as well just call os.path.join directly there without creating those variables here.